source/bluetooth/MicroBitPartialFlashingService.cpp

```diff
@@ -289,16 +349,40 @@ void MicroBitPartialFlashingService::partialFlashingEvent(MicroBitEvent e)
        
        KeyValuePair* flashIncomplete = storage.get("flashIncomplete");
        if(flashIncomplete == NULL){
+
          uint8_t flashIncompleteVal = 0x01;
          storage.put("flashIncomplete", &flashIncompleteVal, sizeof(flashIncompleteVal));
+            
+         // Check if FS exists
+         if(micropython_fs_end != 0x00) {
+            for(uint32_t *page = (uint32_t *)micropython_fs_start; page < (uint32_t *)(micropython_fs_end - MICROBIT_CODEPAGESIZE); page += (MICROBIT_CODEPAGESIZE / sizeof(uint32_t))) {
```

Comment:
I'm not sure why but I'm having to skip the final page of the filesystem
After partial flashing the script won't run. Looking at the page before/after flashing there's a single byte different at `0x72000`

@microbit-carlos is that an important flag for MicroPython? I'm not sure why it's there to start off with, but it seems important!

```
diff pre.hex post.hex
2c2
< :10200000FDFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE2
---
> :10200000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE0
```

Reply:
That byte is the indicator of the "persistent page" (a scratch page for the filesystem): https://github.com/micropython/micropython/blob/b0932fcf2e2f9a81abf7737ed4b2573bd9ad4a49/ports/nrf/modules/uos/microbitfs.c#L79

With the microbit-fs library we place it as the last page in flash, which would be 0x7200 as right now the fs ends in 0x73000 https://github.com/microbit-foundation/micropython-microbit-v2/blob/master/src/codal_port/filesystem.ld#L3

Reply:
Ah yeah, I remember now

So is it best for partial flashing to erase pages up to: 'end of fs' - 1 page, to avoid unecessary erases

Partial flashing should have put the 0xFD in regardless, so I think there's a bug there which I'll have to look at anyway

Reply:
This page could actually be placed at the top or bottom of the filesystem flash area, so the ` 'end of fs' - 1` rule is not a guarantee.
Nevertheless, as you mentioned, the hex file from a Python Editor should include that byte anyway, so partial flashing should detect that page as being the same and not touch it?

Reply:
Ok cool, I'll make sure to erase the full region then
Yeah, it should be flashed so I'll check whats up with that